### DISCUSSION TITLE
A tree-sitter parser is help-wanted

### DISCUSSION THREAD

**Discussion by Freed-Wu** [neomutt/mutt-language-server#3] [Announcements]
In order to realize diagnostic like [this language server](https://github.com/termux/termux-language-server), a tree-sitter parser is needed. A WIP repo is [here](https://github.com/neomutt/tree-sitter-muttrc).
However, I am not familiar with writing a parser and muttrc is a little complex. Any little help from anyone who are good at it or want to take a try is acknowledged.

**Comment by Freed-Wu** [comment]
![Screenshot from 2024-02-14 09-50-29](https://github.com/neomutt/mutt-language-server/assets/32936898/51b084c0-e1b3-4e62-be68-f94981853de7)

The above is tree-sitter, the below is vim syn.

**Comment by Freed-Wu** [comment]
@flatcap I got bug report due to some upstream library's upgrade (pygls: 1.X.Y -> 2.0.0). However,
I lose the permission to [mutt-language-server](https://github.com/neomutt/mutt-language-server) and related projects [tree-sitter-muttrc](https://github.com/neomutt/tree-sitter-muttrc)
[lsp-tree-sitter](https://github.com/neomutt/lsp-tree-sitter) :cry:

**Comment by flatcap** [comment]
@Freed-Wu
Sorry about that.  I've re-invited you back to NeoMutt.

About once a year, I sort through all the users and clear out people who I haven't heard from in over six months.
It's a delicate balancing act between:
- Invite everyone who contributes to make them feel welcome and encourage more help
- Clear out the non-contributors, so that the members reflect how many people are actually doing any work

**Reply by Freed-Wu** [reply]
> @flatcap I got bug report due to some upstream library's upgrade (pygls: 1.X.Y -> 2.0.0). However, I lose the permission to [mutt-language-server](https://github.com/neomutt/mutt-language-server) and related projects [tree-sitter-muttrc](https://github.com/neomutt/tree-sitter-muttrc) [lsp-tree-sitter](https://github.com/neomutt/lsp-tree-sitter) 😢

Just give me the permissions of the above repos is enough. Because neomuttrc's syntax is stable, so these repos will not occur any large changes in the future.
